packages/docusaurus-theme-classic/src/theme/ScrollMonitorProvider/index.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useMemo, useRef} from 'react';
+
+import ScrollMonitorContext from '@theme/ScrollMonitorContext';
+import type {Props} from '@theme/ScrollMonitorProvider';
+
+function ScrollMonitorProvider(props: Props): JSX.Element {
+  const isScrollMonitorEnabledRef = useRef(true);
+
+  const handlers = useMemo(
+    () => ({
+      enableScrollMonitor: () => {
+        isScrollMonitorEnabledRef.current = true;
+      },
+      disableScrollMonitor: () => {
+        isScrollMonitorEnabledRef.current = false;
+      },
+    }),
+    [],
+  );
+
+  return (
+    <ScrollMonitorContext.Provider
+      value={{...handlers, isScrollMonitorEnabledRef}}>
```

Comment:
`value={{ ... }}` creates a new context value on each re-render, leading to subscribers re-rendering too Even if effects don't re-execute it's not the best.

Why not add your ref to `useMemo` too so that context value remains stable?

Reply:
Good point, will do